lib/s3middleware/lifecycleHelpers/LifecycleUtils.ts

```diff
@@ -1,13 +1,12 @@
 import assert from 'assert';
 
 import LifecycleDateTime from './LifecycleDateTime';
-import { supportedLifecycleRules } from '../../constants';
 
 export default class LifecycleUtils {
     _supportedRules: string[];
     _datetime: LifecycleDateTime;
 
-    constructor(supportedRules?: string[], datetime?: LifecycleDateTime) {
+    constructor(supportedRules: string[], datetime?: LifecycleDateTime) {
```

Comment:
If we remove the `?` to say we want it to be defined, then we should add an assert in the constructor to ensure it is, the existing one only test if the variable is an array, if not a falsy value